### PR TITLE
PCHR-2160: Add wrapper service to check permissions

### PR DESCRIPTION
**Pull request by dhavalsharma**
## Overview
This PR creates a new service to wrap check if currently logged in user is admin or not for now. 

In future other endpoints can be added to enhance the permissions check.

## Technical Details
Creates a new angular service to wrap calls to CRM.checkPerm.

## Comments
None

---

- [X ] Tests Pass

### CODE REVIEW

File: org.civicrm.reqangular/src/common/services/check-perm.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+define([
+  'common/modules/services'
+], function (services) {
+  'use strict';
+
+  (function (CRM) {
+    services.factory('CheckPermissions', function () {
+
+      /**
```
Comment (deb1990) [inline org.civicrm.reqangular/src/common/services/check-perm.js]: Comment in wrong place
```javascript
 return {
        /**
         * Checks if currently logged in user has admin access
         */
        canAdmin: function () {
          return CRM.checkPerm('CiviHRLeaveAndAbsences: Administer Leave and Absences');
        }
      };
```

File: org.civicrm.reqangular/src/common/services/check-perm.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+       * Checks if currently logged in user has admin access
+       */
+      return {
+        canAdmin: function () {
+          return CRM.checkPerm('CiviHRLeaveAndAbsences: Administer Leave and Absences');
+        }
+      }
```
Comment (deb1990) [inline org.civicrm.reqangular/src/common/services/check-perm.js]: Semicolon missing

File: org.civicrm.reqangular/src/common/services/check-perm.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    })
```
Comment (deb1990) [inline org.civicrm.reqangular/src/common/services/check-perm.js]: Semicolon missing

File: org.civicrm.reqangular/test/services/check-perm_test.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
+(function (CRM) {
+  define([
+    'common/angular',
+    'common/angularMocks',
```
Comment (deb1990) [inline org.civicrm.reqangular/test/services/check-perm_test.js]: Do we need mocks?

File: org.civicrm.reqangular/test/services/check-perm_test.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+      beforeEach(inject(function (_CheckPermissions_) {
+        checkPermissions = _CheckPermissions_;
+      }));
+
+      describe('canAdmin', function () {
+        var isAdmin;
+
+        beforeEach(function () {
+          spyOn(CRM, 'checkPerm').and.returnValue(true);
```
Comment (deb1990) [inline org.civicrm.reqangular/test/services/check-perm_test.js]: Better setup to make the test more appropriate
```javascript
spyOn(CRM, 'checkPerm').and.callFake(returnTrueIfPermissionIs('CiviHRLeaveAndAbsences: Administer Leave and Absences'));

function returnTrueIfPermissionIs(permissionToCheck) {
  return function(permission) {
    return (permission === permissionToCheck);
  }
}

```

File: org.civicrm.reqangular/test/services/check-perm_test.js (hunk 4 above)
Comment (dhavalsharma) [inline org.civicrm.reqangular/test/services/check-perm_test.js]: removing it gives `ReferenceError: module is not defined`

File: org.civicrm.reqangular/test/services/check-perm_test.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        $rootScope = _$rootScope_;
+        $q = _$q_;
+        checkPermissions = _CheckPermissions_;
+      }));
+
+      describe('canAdmin', function () {
+        var promise;
+
+        beforeEach(function () {
+          spyOn(CRM, 'checkPerm').and.callFake(function () {
```
Comment (deb1990) [inline org.civicrm.reqangular/test/services/check-perm_test.js]: We dont need a promise here, `checkPerm` does not return promise.
`spyOn(CRM, 'checkPerm').and.callFake(returnTrueIfPermissionIs('CiviHRLeaveAndAbsences: Administer Leave and Absences'));` would still work.

**Pull request merged**